plenum/client/client.py

```diff
@@ -382,40 +411,134 @@ def getRepliesFromAllNodes(self, identifier: str, reqId: int):
-                return checkIfMoreThanFSameItems(resultsList, self.f)
-        else:
-            return False  # UNCONFIRMED
+            raise KeyError(full_req_id)
+        proved_reply = self.take_one_proved(replies, full_req_id)
+        if proved_reply:
+            logger.debug("Found proved reply for {}".format(full_req_id))
+            return proved_reply
+        quorumed_reply = self.take_one_quorumed(replies, full_req_id)
+        if quorumed_reply:
+            logger.debug("Reply quorum for {} achieved"
+                         .format(full_req_id))
+            return quorumed_reply
+
+    def take_one_quorumed(self, replies, full_req_id):
+
+        # excluding state proofs from check since they can be different
+        def without_state_proof(result):
+            if STATE_PROOF in result:
+                result.pop('state_proof')
+            return result
+
+        results = [without_state_proof(reply["result"])
+                   for reply in replies.values()]
+
+        first = results[0]
+        if all(result == first for result in results):
+            return first
+        logger.error("Received a different result from "
+                     "at least one node for {}"
+                     .format(full_req_id))
+        return checkIfMoreThanFSameItems(results, self.f)
+
+    def take_one_proved(self, replies, full_req_id):
+        """
+        Returns one reply with valid state proof
+        """
+        for sender, reply in replies.items():
+            result = reply['result']
+            if STATE_PROOF not in result:
+                logger.debug("There is no state proof in "
+                             "reply for {} from {}"
+                             .format(full_req_id, sender))
+                continue
+            if not self.validate_multi_signature(result):
+                logger.warning("{} got reply for {} with bad "
+                               "multi signature from {}"
+                               .format(self.name, full_req_id, sender))
+                # TODO: do something with this node
+                continue
+            if not self.validate_proof(result):
+                logger.warning("{} got reply for {} with invalid "
+                               "state proof from {}"
+                               .format(self.name, full_req_id, sender))
+                # TODO: do something with this node
+                continue
+            return reply
+
+    def validate_multi_signature(self, result):
+        """
+        Validates multi signature
+        """
+        multi_signature = result[STATE_PROOF]['multi_signature']
+        state_root_hash = result[STATE_PROOF]['root_hash']
+        pool_state_root_hash = multi_signature['pool_state_root']
+        participants = multi_signature['participants']
+        signature = multi_signature['signature']
+        full_state_root = state_root_hash + pool_state_root_hash
```

Comment:
It's better to use one utility method in nodes and client to create a final message to be validated